### PR TITLE
[Backport] IdentifyConverter converts NULL-values into NULL-objects

### PR DESCRIPTION
**Pull request by jerrinot**
Backport of #6192

### CODE REVIEW

**Comment by jerrinot** [comment]
@ajermakovics, @serkan-ozal: can you guys have a look? thanks!

File: hazelcast/src/main/java/com/hazelcast/query/impl/UUIDConverter.java [hunk 1]
```diff
@@ -33,6 +33,9 @@ private UUIDConverter() {
 
     @Override
     public Comparable convert(Comparable value) {
+        if (value == null) {
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/query/impl/UUIDConverter.java]: this was another converter where null-value handling was not implemented properly

**Comment by serkan-ozal** [comment]
:+1:

**Comment by ajermakovics** [comment]
:+1:

**Pull request merged**